textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/regexmatch/RegexToGramQueryTranslatorTest.java

```diff
@@ -10,6 +12,16 @@
 
 public class RegexToGramQueryTranslatorTest {
 	
+	private void printTranslatorResult(String regex) {
+		GramBooleanQuery exactQuery = RegexToGramQueryTranslator.translate(regex);
+		
+		System.out.println("regex: "+regex);
```

Comment:
There are a few problems we face in testing the result. 
1, the result contains lots of redundant information. For example, "data*(bcd|pqr)", the result is "(dat AND (dat) AND (pqr OR bcd))". The second "dat" is apparently redundant. The code of simplifying boolean expression is not yet done.
2, even if the result is simplified, one boolean expression has many equivalent forms with different tree structures. The function we wrote works if two have similar structure. We can write a better comparing function, one way I can think of is to write truth tables.
3, even if we get simplified results, and a good comparing function. The expected result we write could be wrong in the first place. When the regex gets complicated, it's hard to manually analyze.
So I'm thinking to run RegexMatcher, with translator enabled/disabled, and compare the results. We need to write massive, comprehensive test cases to catch errors, and I believe our current code has errors, we just haven't found them yet.